pkg/ipamd/ipamd.go

```diff
@@ -583,83 +612,82 @@ func (c *IPAMContext) tryFreeENI() {
 	}
 }
 
-// tryUnassignIPsFromAll determines if there are IPs to free when we have extra IPs beyond the target and warmIPTargetDefined
+// tryUnassignIPsorPrefixesFromAll determines if there are IPs to free when we have extra IPs beyond the target and warmIPTargetDefined
 // is enabled, deallocate extra IP addresses
-func (c *IPAMContext) tryUnassignIPsFromAll() {
-	if _, over, warmIPTargetDefined := c.ipTargetState(); warmIPTargetDefined && over > 0 {
+func (c *IPAMContext) tryUnassignCidrsFromAll() {
+
+	_, over, warmTargetDefined := c.datastoreTargetState()
+
+	//WARM IP targets not defined then check if WARM_PREFIX_TARGET is defined.
+	if !warmTargetDefined {
+		over = c.shouldRemoveExtraPrefixes()
```

Comment:
nit: `shouldRemoveExtraPrefixes` -> `computeOverPrefixes`

Reply:
Renamed it to 1computeExtraPrefixesOverWarmTarget`